src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.cs

```diff
@@ -1510,5 +1510,201 @@ void M([|D|]&lt;int&gt; lambda)
                 context.VerifyResult(metadataAsSourceFile, expected);
             }
         }
+
+        [WorkItem(29786, "https://github.com/dotnet/roslyn/issues/29786")]
+        [Fact, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+        public async Task TestSByteMinValue()
+        {
+            var source = @"
+class C
+{
+    sbyte Goo = sbyte.[|MinValue|];
+}";
+
+            var expected = "public const SByte MinValue = -128;";
+
+            using (var context = TestContext.Create(LanguageNames.CSharp, sourceWithSymbolReference: source))
+            {
+                var navigationSymbol = await context.GetNavigationSymbolAsync();
+                var metadataAsSourceFile = await context.GenerateSourceAsync(navigationSymbol);
+                var document = context.GetDocument(metadataAsSourceFile);
+                var text = await document.GetTextAsync();
+                var line = text.Lines.GetLineFromPosition(metadataAsSourceFile.IdentifierLocation.SourceSpan.Start);
+                var lineText = line.ToString().Trim();
+
+                Assert.Equal(expected, lineText);
+            }
+        }
+
+        [WorkItem(29786, "https://github.com/dotnet/roslyn/issues/29786")]
+        [Fact, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+        public async Task TestSByteMinValueVB()
+        {
+            var source = @"
+Class C
+    Public Goo = SByte.[|MinValue|]
+End Class";
+
+            var expected = "Public Const MinValue As [SByte] = -128";
+
+            using (var context = TestContext.Create(LanguageNames.VisualBasic, sourceWithSymbolReference: source))
+            {
+                var navigationSymbol = await context.GetNavigationSymbolAsync();
+                var metadataAsSourceFile = await context.GenerateSourceAsync(navigationSymbol);
+                var document = context.GetDocument(metadataAsSourceFile);
+                var text = await document.GetTextAsync();
+                var line = text.Lines.GetLineFromPosition(metadataAsSourceFile.IdentifierLocation.SourceSpan.Start);
+                var lineText = line.ToString().Trim();
+
+                Assert.Equal(expected, lineText);
```

Comment:
these tests have a lot of duplication.  can we extract a helper function they all defer to?

Reply:
Sure thing